tests/framework/db/oci/CommandTest.php

```diff
@@ -42,4 +43,36 @@ public function batchInsertSqlProvider()
 
         return $data;
     }
+
+    /**
+     * Testing the "ORA-01461: can bind a LONG value only for insert into a LONG column"
+     *
+     * @return void
+     */
+    public function testCLOBStringInsertion()
+    {
+        $db = $this->getConnection();
+
+        if ($db->getSchema()->getTableSchema('longstring') !== null) {
+            $db->createCommand()->dropTable('longstring')->execute();
+        }
+
+        $db->createCommand()->createTable('longstring', ['message' => Schema::TYPE_TEXT])->execute();
+        $long_data = str_pad('-', 4001, '-=', STR_PAD_LEFT);
+        try {
+            $db->createCommand()->insert('longstring', [
+                'message' => $long_data,
+            ])->execute();
+
+        } catch (\yii\db\Exception $exception) {
```

Comment:
Looks suspicious. Do we expect that another exception would be triggered?

Reply:
the error can be another things like an error during creation of the table, database gone or other stuff.

I made it that way so it's more clear what goes wrong

Reply:
But is it throwing exception during the test?

Reply:
If the query complete successfuly no exception at all
if the test fail, with the ora-01461 a specific message is throw
if the test fail for another reason the test will throwi it normal exception.


The test is specific to string > 4 000, if it fail i want to know if it's because of this test or another reason.

Is my logic good ?

Reply:
Do you mean that the test is unstable and can randomly fail?

Reply:
no the test is reproductible and stable, no exception is leveled during this test because @243083df proposed fix work well

Before the fix this test has leveled an exception, who is catched by :

```php
 $this->assertNotContains('ORA-01461', substr($exception->getMessage(),0,100))
```
if this test fail i just wanted to make it the way i made it normaly you see immediatly if it's caused by the tested scenario or another stuff.

it's better if i made a " $this->fail($e->$exception->getMessage() " in the expcetion part ?

Reply:
If the exception is never thrown under normal circumstances, I think that we should not add `try-catch` at all. Its presence makes reader wonder why it's there.